### ISSUE TITLE
Targeted Universal Adversarial Perturbations not Working

### ISSUE THREAD

**Issue by taylover2016** [Trusted-AI/adversarial-robustness-toolbox#944]
**Describe the bug**
I tried Targeted Universal Adversarial Perturbations on my Neural Network, which worked well with FGM.
Then I switched to Targeted Universal Adversarial Perturbations, but the adversarial output is the same
as the original input.

Here's a segment of my code:

"""
    White-Box Classifier
"""
classifier = PyTorchClassifier(
            model=model,
            input_shape=(data.shape),
            nb_classes=40,
            loss=nn.CrossEntropyLoss(),
            device_type="cpu"
        )

predictions = classifier.predict(data)

accuracy = np.sum(np.argmax(predictions, axis=1) == np.argmax(test_set_y, axis=1)) / test_set_y.shape[0]
print("Accuracy on benign test examples: {}%".format(accuracy * 100))

# Generate adversarial test examples

"""
    White-Box Attacks
"""
# Targeted Universal
attacker_param = {
        "estimator": classifier,
        "norm": "inf", 
        "eps": 0.1,
        "eps_step": 0.1,
        "targeted": False,
        "num_random_init": 0,
        "batch_size": 32,
        "minimal": False
        }

attacker = targeted_universal_perturbation.TargetedUniversalPerturbation(
    classifier=classifier,
    attacker_params=attacker_param
)

label_perturbed = np.eye(test_set_y.shape[0])
np.random.shuffle(label_perturbed)

x_test_adv = attacker.generate(
    x=data,
    y=label_perturbed
    )

# Evaluate the performance
perturbed_predictions = classifier.predict(x_test_adv)

accuracy = np.sum(np.argmax(perturbed_predictions, axis=1) == np.argmax(test_set_y, axis=1)) / test_set_y.shape[0]
print("Accuracy on adversarial test examples: {}%".format(accuracy * 100))

**To Reproduce**

Apply the attack to the classifier and the adversarial output is the same as the original ones.

**Expected behavior**
The Adversarial Output should be different.

**Screenshots**
Some of the console output:
Model is now in the evaluation mode.
The size of the input is:
torch.Size([40, 1, 57, 47])
Accuracy on benign test examples: 87.5%
Accuracy on adversarial test examples: 87.5%

The original predictions are as follows:
<img width="925" alt="截屏2021-03-08 12 16 15" src="https://user-images.githubusercontent.com/53425593/110273694-15df9500-8008-11eb-8656-ec94c18dfe04.png">

And this is the screenshot for the perturbed predictions:
<img width="925" alt="截屏2021-03-08 12 17 25" src="https://user-images.githubusercontent.com/53425593/110273767-3f002580-8008-11eb-996a-309381e2161c.png">

**System information (please complete the following information):**
 - macOS 11.2.2
 - Python 3.8.8
 - ART version 1.5.2
 - PyTorch 1.8.0

**Comment by beat-buesser** [comment]
Hi @taylover2016 Thank you very much for exploring ART. Could you please provide a reproducible example in a Google Colab notebook?

**Comment by taylover2016** [comment]
@beat-buesser Hi, I've provided the necessary code, the dataset and the model together with the params in the latest reply to the issue #943.
Again, Thx for the reply!

**Comment by taylover2016** [comment]
Hello, I've tried the input_shape thing just the same as in #943 , sadly though it didn't work.
The accuracy is still the same as the previous one.
Never mind since Auto-PGD and Square Attack is now working.
I'll go and check SimBA
